packages/next/src/server/lib/router-utils/proxy-request.ts

```diff
@@ -24,11 +24,13 @@ export async function proxyRequest(
     target,
     changeOrigin: true,
     ignorePath: true,
-    xfwd: true,
     ws: true,
     // we limit proxy requests to 30s by default, in development
     // we don't time out WebSocket requests to allow proxying
     proxyTimeout: proxyTimeout === null ? undefined : proxyTimeout || 30_000,
+    headers: {
+      'x-forwarded-host': req.headers.host || '',
```

Comment:
Still set this header, but without concating with existing ones. See https://github.com/vercel/next.js/pull/57815/files#r1378313718